src/node_buffer.cc

```diff
@@ -602,6 +602,110 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 }
 
 
+int32_t IndexOf(char* haystack, size_t h_len, char* needle, size_t n_len) {
```

Comment:
`const char*` for the haystack and the needle?  (And maybe spell out the len variables for clarity.)